tests/suites/test_suite_pkcs7.function

```diff
@@ -0,0 +1,405 @@
+/* BEGIN_HEADER */
+#include "mbedtls/bignum.h"
+#include "mbedtls/pkcs7.h"
+#include "mbedtls/x509.h"
+#include "mbedtls/x509_crt.h"
+#include "mbedtls/x509_crl.h"
+#include "mbedtls/oid.h"
+#include "sys/types.h"
+#include "sys/stat.h"
+#include "unistd.h"
+/* END_HEADER */
+
```

Comment:
`test_suite_pkcs7` fails with `MBEDTLS_MEMORY_BUFFER_ALLOC_C`. Steps to reproduce:

    scripts/config.py set MBEDTLS_MEMORY_BUFFER_ALLOC_C
    scripts/config.py set MBEDTLS_PLATFORM_MEMORY
    CC=gcc cmake .
    make

    msg "test: MBEDTLS_MEMORY_BUFFER_ALLOC_C"
    make test

(You can build without cmake as well, I'm just reproducing what the CI ran.) The CI logs say:
```

      Start 65: pkcs7-suite
65/86 Test #65: pkcs7-suite ............................***Failed    0.00 sec
PKCS7 Signed Data Parse Pass SHA256 #1 ............................ PASS
PKCS7 Signed Data Parse Pass SHA1 #2 .............................. PASS
PKCS7 Signed Data Parse Pass Without CERT #3 ...................... PASS
PKCS7 Signed Data Parse Fail with multiple signers #4 ............. PASS
PKCS7 Signed Data Parse Fail with multiple certs #4 ............... PASS
PKCS7 Signed Data Parse Fail with corrupted cert #5 ............... PASS
PKCS7 Signed Data Parse Fail with corrupted signer info #6 ........ PASS
PKCS7 Signed Data Parse Fail Version other than 1 #7 .............. PASS
PKCS7 Signed Data Parse Fail Encrypted Content #8 ................. PASS
PKCS7 Signed Data Verification Pass SHA256 #9 ..................... 

```

`tests/scripts/all.sh test_ssl_alloc_buffer_and_mfl` fails in the same location:
```
      Start 65: pkcs7-suite
65/86 Test #65: pkcs7-suite ............................***Failed    0.00 sec
PKCS7 Signed Data Parse Pass SHA256 #1 ............................ PASS
PKCS7 Signed Data Parse Pass SHA1 #2 .............................. PASS
PKCS7 Signed Data Parse Pass Without CERT #3 ...................... PASS
PKCS7 Signed Data Parse Fail with multiple signers #4 ............. PASS
PKCS7 Signed Data Parse Fail with multiple certs #4 ............... PASS
PKCS7 Signed Data Parse Fail with corrupted cert #5 ............... PASS
PKCS7 Signed Data Parse Fail with corrupted signer info #6 ........ PASS
PKCS7 Signed Data Parse Fail Version other than 1 #7 .............. PASS
PKCS7 Signed Data Parse Fail Encrypted Content #8 ................. PASS
PKCS7 Signed Data Verification Pass SHA256 #9 ..................... FATAL: mbedtls_free() outside of managed space
```